platform/darwin/src/NSExpression+MGLAdditions.mm

```diff
@@ -27,12 +27,37 @@ @implementation NSExpression (MGLAdditions)
         return { std::string([(NSString *)value UTF8String]) };
     } else if ([value isKindOfClass:NSNumber.class]) {
         NSNumber *number = (NSNumber *)value;
-        if((strcmp([number objCType], @encode(int))) == 0) {
+        // handle all types defined in NSNumber designated initializers
+        if (strcmp([number objCType], @encode(char)) == 0) {
+            return { number.charValue };
+        } else if (strcmp([number objCType], @encode(unsigned char)) == 0) {
+            return { number.unsignedCharValue };
+        } else if (strcmp([number objCType], @encode(short)) == 0) {
+            return { number.shortValue };
+        } else if (strcmp([number objCType], @encode(unsigned short)) == 0) {
+            return { number.unsignedShortValue };
+        } else if (strcmp([number objCType], @encode(int)) == 0) {
             return { number.intValue };
-        } else if ((strcmp([number objCType], @encode(double))) == 0) {
+        } else if (strcmp([number objCType], @encode(unsigned int)) == 0) {
+            return { number.unsignedIntValue };
+        } else if (strcmp([number objCType], @encode(long)) == 0) {
+            return { number.longValue };
+        } else if (strcmp([number objCType], @encode(unsigned long)) == 0) {
+            return { number.unsignedLongValue };
+        } else if (strcmp([number objCType], @encode(long long)) == 0) {
+            return { number.longLongValue };
+        } else if (strcmp([number objCType], @encode(unsigned long long)) == 0) {
+            return { number.unsignedLongLongValue };
+        } else if (strcmp([number objCType], @encode(float)) == 0) {
+            return { number.floatValue };
+        } else if (strcmp([number objCType], @encode(double)) == 0) {
             return { number.doubleValue };
-        } else if ((strcmp([number objCType], @encode(char))) == 0) {
+        } else if (strcmp([number objCType], @encode(bool)) == 0) {
```

Comment:
As best I can tell, https://github.com/mapbox/mapbox-gl-native/commit/7e208c46cebe5b49d4b7ead9826bbe4229ebabd3 is actually ok and we want `... if ((strcmp([number objCType], @encode(char))) == 0)`.

We used to compare `[number objCType]` (returns `c` for boolean on both 32 and 64 bit system to `@encode(bool)` which returns `c` on 32 bit and `b` on 64 bit. `7e208c4` changed to compare `[number objCType]` to `@encode(char)` which always returns `c`.